backend/internal/system/crypto/hash/service.go

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Package hash provides generic hashing utilities for sensitive data.
+package hash
+
+import (
+	"crypto/pbkdf2"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"sync"
+
+	"github.com/asgardeo/thunder/internal/system/config"
+	"github.com/asgardeo/thunder/internal/system/log"
+)
+
+const (
+	defaultSaltSize         = 16
+	defaultPBKDF2Iterations = 600000
+	defaultPBKDF2KeySize    = 32
+)
+
+var (
+	instance HashService
+	once     sync.Once
+	logger   = log.GetLogger().With(log.String(log.LoggerKeyComponentName, "HashService"))
+)
+
+// HashService defines the interface for hashing services.
+type HashService interface {
```

Comment:
```suggestion
type HashServiceInterface interface {
```
To align with existing naming conversions